tests/scipy_stats_test.py

```diff
@@ -913,5 +913,36 @@ def evaluate_kde(kde, x):
     tree_util.tree_map(lambda a, b: self.assertAllClose(a, b), kde, kde2)
     self.assertAllClose(evaluate_kde(kde, x), kde.evaluate(x))
 
+  @jtu.sample_product(
+    [dict(shape=shape, dtype=dtype, axis=axis, contains_nans=contains_nans)
+      for dtype in jtu.dtypes.integer + jtu.dtypes.floating
+      for shape, axis in (
+        ((0,), None),
+        ((7,), None),
+        ((7,), 0),
+        ((47, 8), None),
+        ((47, 8), 0),
+        ((47, 8), 1),
+        ((10, 5, 21), None),
+        ((10, 5, 21), 0),
+        ((10, 5, 21), 1),
+        ((10, 5, 21), 2),
+      )
+      for contains_nans in [True, False]
+    ])
```

Comment:
To avoid generating a list of all possible combinations, use something like this:
```python
@jtu.sample_product(
    [dict(shape=shape, axis=axis)
      for shape, axis in (
        ((0,), None),
        ((7,), None),
        ((7,), 0),
        ((47, 8), None),
        ((47, 8), 0),
        ((47, 8), 1),
        ((10, 5, 21), None),
        ((10, 5, 21), 0),
        ((10, 5, 21), 1),
        ((10, 5, 21), 2),
      )
    ],
    dtype=jtu.dtypes.integer + jtu.dtypes.floating,
    contains_nans=[True, False])
```